runtime/oti/j9protos.h

```diff
@@ -1133,6 +1133,26 @@ extern J9_CFUNC struct J9Method* jitResolveStaticMethodRef(J9VMThread *vmStruct,
 extern J9_CFUNC struct J9Method* jitResolveSpecialMethodRef(J9VMThread *vmStruct, J9ConstantPool *constantPool, UDATA cpOrSplitIndex, UDATA resolveFlags);
 extern J9_CFUNC struct J9Class * jitGetDeclaringClassOfROMField(J9VMThread *vmStruct, J9Class *clazz, J9ROMFieldShape *romField);
 
+typedef struct J9MethodFromSignatureWalkState {
+	U_8* className;
+	U_16 classNameLength;
+	U_8* methodName;
+	U_16 methodNameLength;
+	U_8* methodSig;
+	U_16 methodSigLength;
+	struct J9VMThread* vmThread;
+	struct J9ClassLoaderWalkState classLoaderWalkState;
+} J9MethodFromSignatureWalkState;
+
+extern J9_CFUNC void
+jitAllMethodFromSignatureEndDo(J9MethodFromSignatureWalkState* state);
+
+extern J9_CFUNC struct J9Method*
+jitAllMethodFromSignatureNextDo(J9MethodFromSignatureWalkState* state);
+
+extern J9_CFUNC struct J9Method*
+jitAllMethodFromSignatureStartDo(J9MethodFromSignatureWalkState* state, J9JavaVM* vm, UDATA flags, U_8* className, U_16 classNameLength, U_8* methodName, U_16 methodNameLength, U_8* methodSig, U_16 methodSigLength);
```

Comment:
@keithc-ca / @pshipton subscribing you guys for a review here. I wanted to get your blessing first for the implementation, following which I will write proper documentations for all the new APIs introduced. I placed them here because I plan to make use of them in a future work item.

Could you review the new APIs? There is an example use of them in JitDump.cpp as part of commit 1e3a722.

Reply:
This API looks reasonable to me, with some suggestions:
* use `J9JNINameAndSignature nameAndSig;` instead of the `method*` fields
* why the `jit` prefix? this may be more generally useful

Reply:
I agree on both fronts. Fixed in 2b02a6f